sympy/solvers/solveset.py

```diff
@@ -896,6 +897,14 @@ def solveset(f, symbol=None, domain=S.Complexes):
     if not isinstance(f, (Expr, Number)):
         raise ValueError("%s is not a valid SymPy expression" % (f))
 
+    if (isinstance(symbol, Function) and not symbol.has(Symbol)):
+        for arg in f.args:
+            if not (arg.has(symbol) or arg.is_Number):
+                raise ValueError("%s is not a valid SymPy expression" % (f))
```

Comment:
It doesn't mean that it is not a valid expression, rather it means that it is not solvable using this method and should be passed to other methods.

Reply:
How do I implement this? How do I pass it to other solvers? @ylemkimon

Reply:
Do not raise a error or return.